Mathlib/RingTheory/GradedAlgebra/HomogeneousLocalization.lean

```diff
@@ -341,14 +356,12 @@ instance : SMul α (HomogeneousLocalization 𝒜 x) where
     simp only [num_smul, den_smul]
     convert congr_arg (fun z : at x => m • z) h <;> rw [Localization.smul_mk]
 
+@[simp] lemma mk_smul (i : NumDenSameDeg 𝒜 x) (m : α) : mk (m • i) = m • mk i := rfl
+
 @[simp]
-theorem smul_val (n : α) (y : HomogeneousLocalization 𝒜 x) : (n • y).val = n • y.val := by
-  induction y using Quotient.inductionOn
-  change Localization.mk _ _ = n • Localization.mk _ _
-  dsimp only
-  rw [Localization.smul_mk]
-  congr 1
-#align homogeneous_localization.smul_val HomogeneousLocalization.smul_val
+theorem val_smul (n : α) : ∀ y : HomogeneousLocalization 𝒜 x, (n • y).val = n • y.val :=
```

Comment:
The PR looks very good to me! I am just not sure about the need to rename `smul_val` as `val_smul`, etc?

Reply:
The term is actually `HomogeneousLocalization.val (n • y)` and it should be `val_smul` according to the naming convention. This also matches other API lemmas of the same form such as `SetLike.val_smul`.